db/sqlglue.c

```diff
@@ -1342,30 +1342,41 @@ static int using_old_style_name(char *namebuf, int len, struct schema *schema,
     return stat1_find(namebuf, schema, db, ixnum, trans);
 }
 
+#define SNPRINTF(str, size, fmt, ...)                                          \
+    {                                                                          \
+        ret = snprintf(str, size, fmt, __VA_ARGS__);                           \
+        if (ret >= size)                                                       \
```

Comment:
We dont do current += size here, which means that we wrote size bytes, but when we goto done to compute crc we wont do this field. Dont know if you want to include this field.